src/mappers/small-catches.mapper.js

```diff
@@ -5,6 +5,82 @@ import {
 } from '../utils/date-utils.js'
 import { getBaseUrl } from '../utils/url-utils.js'
 
+/**
+ * Maps an array of count objects to a new format, extracting the `method_id` and optionally
+ * adding a `small_catch_id` property.
+ *
+ * @param {Array<{ count: number, method: string }>} countsArray - An array of objects, each containing a `count` and `method`.
+ * @param {string} [smallCatchId] - The optional small catch ID to be included in each mapped object.
+ * @returns {Array<{ count: number, method_id: string, small_catch_id?: string }>} A new array of objects containing `count`, `method_id`,
+ * and optionally `small_catch_id`.
+ *
+ * @example
+ * const countsArray = [
+ *   { count: 3, method: 'method/1' },
+ *   { count: 2, method: 'method/2' },
+ *   { count: 1, method: 'method/3' }
+ * ];
+ * const smallCatchId = '123';
+ * const result = mapCounts(countsArray, smallCatchId);
+ * // [
+ * //   { count: 3, method_id: '1', small_catch_id: '123' },
+ * //   { count: 2, method_id: '2', small_catch_id: '123' },
+ * //   { count: 1, method_id: '3', small_catch_id: '123' }
+ * // ]
+ */
```

Comment:
I don't think we need the @example, here or below; that's what the unit tests will show